### PR TITLE
wonderdraft: init at 1.1.7.3

### PR DESCRIPTION
**Pull request by IGJoshua**
## Description of changes
This PR adds a package for the dungeon mapping tool Wonderdraft by Megasploot.

I'm adding myself as a maintainer for this package, I hope that's alright. I also have another PR for Dungeondraft. #253034

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — CHANGES_REQUESTED** [review]
The part in `maintainer-list.nix` needs to be in a separate commit. You can find good examples of the commit message in the history of that file https://github.com/NixOS/nixpkgs/commits/master/maintainers/maintainer-list.nix .

**Comment by IGJoshua** [comment]
I've made updates to this PR, I think it's ready for consideration again, pending the checks.

File: pkgs/applications/graphics/wonderdraft/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+, dpkg
+, xorg
+, libGL
+, alsa-lib
+, pulseaudio
+, fetchurl
+}:
+
+stdenv.mkDerivation rec {
+  name = "wonderdraft-${version}";
```
Comment (nagy) [inline pkgs/applications/graphics/wonderdraft/default.nix:13]: Use `pname = "wonderdraft";`

File: pkgs/applications/graphics/wonderdraft/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}" \
+      $out/opt/Wonderdraft/Wonderdraft.x86_64
+  '';
+
+  meta = with lib; {
+    homepage = "https://wonderdraft.net/";
+    description = "A mapmaking tool for Tabletop Roleplaying Games, designed for city, region, or world scale.";
+    license = licenses.unfree;
+    platforms = platforms.linux;
```
Comment (nagy) [inline pkgs/applications/graphics/wonderdraft/default.nix:57]: I think `platforms = [ "x86_64-linux" ];` is more appropriate.

File: pkgs/applications/graphics/wonderdraft/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    maintainers = with maintainers; [ jsusk ];
```
Comment (nagy) [inline pkgs/applications/graphics/wonderdraft/default.nix:58]: Can you add `sourceProvenance = with sourceTypes; [ binaryNativeCode ];`?

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/wonderdraft/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    sha256 = "157himdhpijcc7dk696p6d38kdsk8x44jzagraa8ax1mkhbikilb";
+  };
+  sourceRoot = ".";
+  unpackCmd = "${dpkg}/bin/dpkg-deb -x $curSrc .";
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (nagy) [inline pkgs/applications/graphics/wonderdraft/default.nix:28]: I forgot to say that this needs `runHook preInstall` at the beginning and `runHook postInstall` at the end. After that, its fine.

**Review by nagy — CHANGES_REQUESTED** [review]

**Review by nagy — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -36706,6 +36706,8 @@ with pkgs;
 
   wofi-emoji = callPackage ../applications/misc/wofi-emoji { };
 
+  wonderdraft = callPackage ../applications/graphics/wonderdraft {  };
```
Comment (kip93) [inline pkgs/top-level/all-packages.nix:36709]: Please use the new by-name structure.

Comment (wegank) [reply pkgs/top-level/all-packages.nix]: Done.

**Review by marsam — APPROVED** [review]

**Pull request merged**